addons/survey/models/survey_survey.py

```diff
@@ -397,6 +408,13 @@ def _onchange_session_speed_rating(self):
             survey.question_ids._update_time_limit_from_survey(
                 is_time_limited=survey.session_speed_rating, time_limit=survey.session_speed_rating_time_limit)
 
+    @api.onchange('restrict_user_ids')
+    def _onchange_restrict_user_ids(self):
```

Comment:
Don't we need the counterpart when `userId` changes and there are already `restrict_user_ids`?

Reply:
yes we shoud. What about this?